src/modrover/rover.py

```diff
@@ -38,6 +41,8 @@ def __init__(
 
         self.learners: dict[LearnerID, Learner] = {}
 
+        self.super_learner = None
```

Comment:
We can actually make `super_learner` a property with `_super_learner` as the private attribute, so that user won't cannot modify it